app/controllers/Checkout.scala

```diff
@@ -32,33 +36,43 @@ object Checkout extends Controller with LazyLogging with ActivityTracking {
     val IdentityGuestPasswordSettingToken = "newSubs_token"
   }
 
-  def zuoraService(implicit res: TouchpointBackend.Resolution): CatalogService =
-    res.backend.catalogService
 
   def checkoutService(implicit res: TouchpointBackend.Resolution): CheckoutService =
     res.backend.checkoutService
 
-  def renderCheckout = NoCacheAction.async { implicit request =>
-    implicit val touchpointBackend = TouchpointBackend.forRequest(PreSigninTestCookie, request.cookies)
-    trackAnon(CheckoutReachedActivity("United Kingdom"))
+  def renderCheckout(countryGroup: CountryGroup) = NoCacheAction.async { implicit request =>
 
-    val authUserOpt = authenticatedUserFor(request)
+    implicit val resolution: TouchpointBackend.Resolution = TouchpointBackend.forRequest(PreSigninTestCookie, request.cookies)
+    implicit val tpBackend = resolution.backend
 
-    def fillForm(): Future[Form[SubscriptionData]] = for {
-      fullUserOpt <- authUserOpt.fold[Future[Option[IdUser]]](Future.successful(None))(au => IdentityService.userLookupByCredentials(au.credentials))
-    } yield {
-        fullUserOpt.map { idUser =>
-          SubscriptionsForm().fill(SubscriptionData.fromIdUser(idUser))
-        } getOrElse {
-          SubscriptionsForm()
-        }
-      }
-    for {
-      filledForm <- fillForm()
-      products <- zuoraService.products
-      selectedProduct = products.find(p => p.frequency == BillingFrequency.Month).get
-    } yield {
-      Ok(views.html.checkout.payment(filledForm, userIsSignedIn = authUserOpt.isDefined, products, selectedProduct, touchpointBackend))
+    trackAnon(CheckoutReachedActivity("United Kingdom"))
+    val authenticatedUser = authenticatedUserFor(request)
+
+    val subscriptionData = (for {
+      authUser <- OptionT(Future.successful(authenticatedUser))
+      idUser <- OptionT(IdentityService.userLookupByCredentials(authUser.credentials))
+    } yield SubscriptionData.fromIdUser(idUser)).run
+
+    val plans = catalog.planMap.values.toSeq
+    val defaultPlan = catalog.digipackMonthly
+
+    subscriptionData map { subsData =>
+      val form = subsData.fold(SubscriptionsForm()) { data => SubscriptionsForm().fill(data) }
+      val countryGroupWithDefault =
+        subsData.flatMap { data => CountryGroup.byCountryNameOrCode(data.personalData.address.countryName) }
+                .getOrElse(countryGroup)
+      val desiredCurrency = countryGroupWithDefault.currency
+      val supportedCurrencies = defaultPlan.pricing.underlying.keySet
+      val currency = if (supportedCurrencies.contains(desiredCurrency)) desiredCurrency else GBP
+
```

Comment:
Just noticing that in both membership and here this whitelisting logic is repeated twice between form rendering and form processing. We should push it into commons in a later PR